core/dictgen/src/rootcling_impl.cxx

```diff
@@ -2236,6 +2236,51 @@ static bool GenerateAllDict(TModuleGenerator &modGen, clang::CompilerInstance *c
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Includes all headers in the given module from this interpreter.
+static void IncludeModuleHeaders(TModuleGenerator &modGen, clang::Module *module, cling::Interpreter &interpreter)
+{
+   // Make a list of modules and submodules that we can check for headers.
+   // We use a SetVector to prevent an infinite loop in unlikely case the
+   // modules somehow are messed up and don't form a tree...
+   llvm::SetVector<clang::Module *> modules;
+   modules.insert(module);
+   for (size_t i = 0; i < modules.size(); ++i) {
+      clang::Module *M = modules[i];
+      for (clang::Module *subModule : M->submodules())
+         modules.insert(subModule);
+   }
+   // List of includes the interpreter has to parse.
+   std::stringstream includes;
+   // Now we include all header files from the previously collected modules.
+   // FIXME: This might hide that we have a missing include if one of the
+   // previous headers already includes this. We should reuse the clang way
+   // of doing this and 'import' each header in its own submodule, then let
+   // the visibility of the decls handle this situation nicely.
+   for (clang::Module *module : modules) {
+      // Iterate over all header types in a module.
+      // FIXME: We currently have to hardcode '4' to do this. Maybe we
+      // will have a nicer way to do this in the future.
+      // NOTE: This is on purpose '4', not '5' which is the size of the
```

Comment:
Maybe creatin a constant with a value 4 will make the code a little more readable.

Reply:
Done, refactored and added a static_assert to prevent any out of bounds when this array changes.